### PR TITLE
✨ Add `GET /user/:wallet/stats` endpoint

### PR DESCRIPTION
**Pull request by WeiJun0827**

### CODE REVIEW

File: src/util/api/likernft/user.js [hunk 1]
```diff
@@ -34,4 +37,36 @@ export async function filterOwnedClassIds(iscnDocs, wallet) {
...
+export async function getUserStat(wallet) {
+  const { data: userStat } = await axios.get(`${COSMOS_LCD_INDEXER_ENDPOINT}/likechain/likenft/v1/user-stat?user=${wallet}`);
+  const {
+    collected_classes: collectedClasses,
+    created_count: createdClassCount,
+    collector_count: createdCollectorCount,
+  } = userStat;
+  const collectedClassCount = collectedClasses.length;
+  const collectedClassIds = collectedClasses.map(c => c.class_id);
+  const arrays = [];
```
Comment (nwingt) [inline src/util/api/likernft/user.js]:
```suggestion
  const batches = [];
```

File: src/util/api/likernft/user.js [hunk 2]
```diff
@@ -34,4 +37,36 @@ export async function filterOwnedClassIds(iscnDocs, wallet) {
...
+
+  const collectedNftValue = collectedClasses.reduce((acc, c) => {
+    const price = priceMap[c.class_id];
+    return price ? acc + price * c.count : acc;
+  }, 0);
+  return {
+    collectedClassCount,
+    collectedNftValue,
+    createdClassCount,
+    createdCollectorCount,
```
Comment (nwingt) [inline src/util/api/likernft/user.js]: For consistency 
### A
```suggestion
    collectedNFTClassCount,
    collectedNFTValue,
    createdNFTClassCount,
    createdCollectorCount,
```

### B
```suggestion
    collectedClassCount,
    collectedValue,
    createdClassCount,
    createdCollectorCount,
```

File: src/util/api/likernft/user.js [hunk 3]
```diff
@@ -34,4 +37,36 @@ export async function filterOwnedClassIds(iscnDocs, wallet) {
...
+
+  const priceMap = {};
+  docs.forEach((doc) => {
+    const { classId, currentPrice } = doc.data();
+    priceMap[classId] = currentPrice;
+  });
+
+  const collectedNftValue = collectedClasses.reduce((acc, c) => {
+    const price = priceMap[c.class_id];
+    return price ? acc + price * c.count : acc;
```
Comment (nwingt) [inline src/util/api/likernft/user.js]: I think we can also calculate `collectedCount` (sum of `c.count`) here, which maybe useful in the future 😊

File: src/util/api/likernft/user.js [hunk 4]
```diff
@@ -34,4 +37,40 @@ export async function filterOwnedClassIds(iscnDocs, wallet) {
...
+
+  const priceMap = {};
+  docs.forEach((doc) => {
+    const { classId, currentPrice } = doc.data();
+    priceMap[classId] = currentPrice;
+  });
+
+  let collectedCount = 0;
+  let collectedValue = 0;
+  collectedClasses.foreach((c) => {
```
Comment (williamchong) [inline src/util/api/likernft/user.js]: is `foreach` intentionally or is it `forEach`?

**Review by williamchong — CHANGES_REQUESTED** [review]

File: src/util/api/likernft/user.js (hunk 4 above)
Comment (WeiJun0827) [reply src/util/api/likernft/user.js]: It's a typo but somehow it doesn't break the code. Weird...

**Review by williamchong — APPROVED** [review]

**Pull request merged**